Mathlib/CategoryTheory/Sums/Basic.lean

```diff
@@ -203,15 +196,69 @@ theorem sum_obj_inr (F : A ⥤ B) (G : C ⥤ D) (c : C) : (F.sum G).obj (inr c)
   rfl
 
 @[simp]
-theorem sum_map_inl (F : A ⥤ B) (G : C ⥤ D) {a a' : A} (f : inl a ⟶ inl a') :
-    (F.sum G).map f = F.map f :=
+theorem sum_map_inl (F : A ⥤ B) (G : C ⥤ D) {a a' : A} (f : a ⟶ a') :
+    (F.sum G).map ((Sum.inl_ _ _).map f) = (Sum.inl_ _ _).map (F.map f) :=
   rfl
 
 @[simp]
-theorem sum_map_inr (F : A ⥤ B) (G : C ⥤ D) {c c' : C} (f : inr c ⟶ inr c') :
-    (F.sum G).map f = G.map f :=
+theorem sum_map_inr (F : A ⥤ B) (G : C ⥤ D) {c c' : C} (f : c ⟶ c') :
+    (F.sum G).map ((Sum.inr_ _ _).map f) = (Sum.inr_ _ _).map (G.map f) :=
   rfl
 
+section
+
+variable {F G: A ⊕ B ⥤ C}
+  (e₁ : Sum.inl_ _ _ ⋙ F ≅ Sum.inl_ _ _ ⋙ G)
+  (e₂ : Sum.inr_ _ _ ⋙ F ≅ Sum.inr_ _ _ ⋙ G)
+
+/-- A functor out of a sum is uniquely characterized by its precompositions with `inl_` and `inr_`.
+-/
+def sumIsoExt : F ≅ G :=
+  NatIso.ofComponents (fun x ↦
+    match x with
+    | inl x => e₁.app x
+    | inr x => e₂.app x)
+    (fun {x y} f ↦ by
+      cases f
+      · simpa using e₁.hom.naturality _
+      · simpa using e₂.hom.naturality _)
+
+@[simp]
+lemma sumIsoExt_hom_app_inl (a : A) : (sumIsoExt e₁ e₂).hom.app (inl a) = e₁.hom.app a := rfl
+
+@[simp]
+lemma sumIsoExt_hom_app_inr (b : B) : (sumIsoExt e₁ e₂).hom.app (inr b) = e₂.hom.app b := rfl
+
+@[simp]
+lemma sumIsoExt_inv_app_inl (a : A) : (sumIsoExt e₁ e₂).inv.app (inl a) = e₁.inv.app a := rfl
+
+@[simp]
+lemma sumIsoExt_inv_app_inr (b : B) : (sumIsoExt e₁ e₂).inv.app (inr b) = e₂.inv.app b := rfl
+
+end
+
+section
+
+variable (F : A ⊕ B ⥤ C) (a : A) (b : B)
```

Comment:
Similar style point: I'd actually move the `(a : A) (b : B)` below the `def`, so no human elaboration is required to know that they are not used by the `def`.